custom_components/opnsense/pyopnsense/__init__.py

```diff
@@ -853,193 +855,195 @@ def send_wol(self, interface, mac):
+        if not match:
+            raise ValueError("Invalid uptime format")
+        days_str, hours_str, minutes_str, seconds_str = match.groups()
+        days: int = int(days_str) if days_str else 0
+        hours: int = int(hours_str)
+        minutes: int = int(minutes_str)
+        seconds: int = int(seconds_str)
+        system["uptime"] = days * 86400 + hours * 3600 + minutes * 60 + seconds
+
+        boottime: datetime = datetime.now() - timedelta(seconds=system["uptime"])
+        system["boottime"] = boottime.timestamp()        
+        load_str: str = time_info["loadavg"]
+        load_list: list[str] = load_str.split(", ")
+        system["load_average"] = {
+            "one_minute": float(load_list[0]),
+            "five_minute": float(load_list[1]),
+            "fifteen_minute": float(load_list[2]),
+        }
+        _LOGGER.debug(f"[get_telemetry_system] system: {system}")
+        return system
 
-foreach ($ovpn_servers as $server) {
-    $vpnid = $server["vpnid"];
-    $name = $server["name"];
-    $conn_count = count($server["conns"]);
-    $total_bytes_recv = 0;
-    $total_bytes_sent = 0;
-    foreach ($server["conns"] as $conn) {
-        $total_bytes_recv += $conn["bytes_recv"];
-        $total_bytes_sent += $conn["bytes_sent"];
-    }
-    
-    $toreturn["openvpn"]["servers"][$vpnid]["name"] = $name;
-    $toreturn["openvpn"]["servers"][$vpnid]["vpnid"] = $vpnid;
-    $toreturn["openvpn"]["servers"][$vpnid]["connected_client_count"] = $conn_count;
-    $toreturn["openvpn"]["servers"][$vpnid]["total_bytes_recv"] = $total_bytes_recv;
-    $toreturn["openvpn"]["servers"][$vpnid]["total_bytes_sent"] = $total_bytes_sent;
-}
+    @_log_errors
+    def _get_telemetry_cpu(self) -> dict:
+        cpu: dict[str, Any] = {}
+        cputype_info: dict[str, Any] = self._post("/api/diagnostics/cpu_usage/getCPUType")
+        _LOGGER.debug(f"[get_telemetry_cpu] cpu_info: {cputype_info}")
+        # Hacky and probably not universally applicable
+        cores_match = re.search(r'\((\d+) cores', cputype_info[0])
+        cpu["count"] = int(cores_match.group(1)) if cores_match else 0
+        # Missing frequency current and max
+        # cpu["frequency"] = {"current": 0, "max": 0}
+        _LOGGER.debug(f"[get_telemetry_cpu] cpu: {cpu}")
+        return cpu
 
-"""
-        data = self._exec_php(script)
+    @_log_errors
+    def _get_telemetry_filesystems(self) -> dict:
+        filesystems: dict[str, Any] = {}
+        filesystems_info: dict[str, Any] = self._post(
+            "/api/diagnostics/system/systemDisk"
+        )
+        _LOGGER.debug(f"[get_telemetry_filesystems] filesystems_info: {filesystems_info}")
+        filesystems = filesystems_info["devices"]
+        # To conform to the previous data being returned
+        for filesystem in filesystems:
+            filesystem["size"] = filesystem.pop("blocks", None)
+            filesystem["capacity"] = f"{filesystem.pop("used_pct")}%"
```

Comment:
I had an issue with VSCode giving me weird errors for this line, had to modify with single quotes:

```python
filesystem["capacity"] = f"{filesystem.pop('used_pct')}%"
```